editor/src/clj/editor/lua_parser.clj

```diff
@@ -192,14 +192,40 @@
                 (when-some [function-name (matching-pred string? (nth var-suffix-node 2))]
                   [module-name function-name arg-exps])))))))))
 
+(defn- deep-search-node [node target-node-type]
+  "Recursively searches for the first occurrence of a specific target node type in the given node tree."
+  (when (seq? node)
+    (or
+      ;; If we find the target node type, return it
+      (when (= target-node-type (first node)) node)
+      ;; Otherwise, recursively search its children
+      (some #(deep-search-node % target-node-type) (rest node)))))
+
+(defn- match-and-extract-args
+  "Checks if the function name matches and extracts argument expressions (:explist) if found."
+  [function-name node check-fn]
+  (when (check-fn)
+    (let [name-and-args-node (deep-search-node node :nameAndArgs)]
+      (when (some? name-and-args-node)
+        (parse-arg-exps name-and-args-node)))))
+
 (defn- matching-functioncall
+  "Searches the node tree recursively for a function call matching the given module-name and function-name.
```

Comment:
It seems like `clj-antlr` misses a lot of data, making it difficult to work with. It's impossible to capture all the function calls as you can in vanilla ANTLR. That’s why I use a search method here, instead of relying on the exact order of nodes. 

Of course, we could rewrite it using raw ANTLR, but that would involve significantly more effort and require rewriting this entire module.

Reply:
But is clj-antlr using generated grammar and parser for Lua? Could we not regenerate the clj part using newer grammar?